ci/travis_before_script_cpp.sh

```diff
@@ -62,7 +62,10 @@ if [ "$ARROW_TRAVIS_USE_TOOLCHAIN" == "1" ]; then
   CMAKE_COMMON_FLAGS="$CMAKE_COMMON_FLAGS -DARROW_WITH_BZ2=ON"
 fi
 
-CMAKE_COMMON_FLAGS="$CMAKE_COMMON_FLAGS -DARROW_MIMALLOC=ON"
+
+if [ "$DISTRO_CODENAME" != "xenial" ]; then
+  CMAKE_COMMON_FLAGS="$CMAKE_COMMON_FLAGS -DARROW_MIMALLOC=ON"
+fi
```

Comment:
This is definitely a hack. Can you instead add an option to toggle on/off mimalloc from .travis.yml?

Reply:
updated. Please take a look